src/components/bottomPanel/tabs/terminal/LogsTerminal.vue

```diff
@@ -32,17 +32,35 @@ const terminalCreated = (
   { terminal, useAutoSize }: ReturnType<typeof useTerminal>,
   root: Ref<HTMLElement | undefined>
 ) => {
-  // `autoCols` is false because we don't want the progress bar in the terminal
-  // to render incorrectly as the progress bar is rendered based on the
-  // server's terminal size.
+  // Track whether server provides terminal size (determined once on first response)
+  // If server provides size, use it for proper progress bar rendering
+  // If not, use auto cols to fill available width
+  let serverProvidesSize: boolean | null = null
+
   // Apply a min cols of 80 for colab environments
   // See https://github.com/comfyanonymous/ComfyUI/issues/6396
-  useAutoSize({ root, autoRows: true, autoCols: false, minCols: 80 })
+  // Start with autoCols: true, will be disabled if server provides size
+  const { resize } = useAutoSize({
+    root,
+    autoRows: true,
+    autoCols: true,
+    minCols: 80
+  })
 
   const update = (entries: Array<LogEntry>, size?: TerminalSize) => {
-    if (size) {
+    // Determine sizing mode once on first response
+    if (serverProvidesSize === null) {
+      serverProvidesSize = size?.cols != null && size.cols > 0
+    }
+
+    // Use server size if provided, otherwise auto sizing handles it
+    if (serverProvidesSize && size?.cols) {
```

Comment:
So if the server starts sending the columns after the first response, we ignore it?